controllers/routing/certificate.go

```diff
@@ -0,0 +1,106 @@
+package routingcontroller
+
+import (
+	"context"
+	"fmt"
+	certmanagerv1 "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+	certmanagermetav1 "github.com/cert-manager/cert-manager/pkg/apis/meta/v1"
+	skiperatorv1alpha1 "github.com/kartverket/skiperator/api/v1alpha1"
+	"github.com/kartverket/skiperator/pkg/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrlutil "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const IstioGatewayNamespace = "istio-gateways"
+
+func (r *RoutingReconciler) SkiperatorRoutingCertRequests(_ context.Context, obj client.Object) []reconcile.Request {
+	certificate, isCert := obj.(*certmanagerv1.Certificate)
+
+	if !isCert {
+		return nil
+	}
+
+	isSkiperatorRoutingOwned := certificate.Labels["app.kubernetes.io/managed-by"] == "skiperator" &&
+		certificate.Labels["skiperator.kartverket.no/controller"] == "routing"
+
+	requests := make([]reconcile.Request, 0)
+
+	if isSkiperatorRoutingOwned {
+		requests = append(requests, reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Namespace: certificate.Labels["application.skiperator.no/app-namespace"],
+				Name:      certificate.Labels["application.skiperator.no/app-name"],
+			},
+		})
+	}
+
+	return requests
+}
+
+func (r *RoutingReconciler) reconcileCertificate(ctx context.Context, routing *skiperatorv1alpha1.Routing) (reconcile.Result, error) {
+	// TODO: Fix controllerProgressing
+
+	var err error
+	//controllerName := "Certificate"
+	//r.SetControllerProgressing(ctx, routing, controllerName)
```

Comment:
IMO, drop this. IIRC this will change the resourceVersion of the object without actually doing anything which can cause race conditions when we try to apply updates to the object. It's better to only update the status after something went wrong/well.